### PR TITLE
cleanups to readme, makefiles, etc.

### PR DESCRIPTION
**Pull request by rshriram**
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,22 +42,30 @@ fmt:
...
 
 .PHONY: clean
 clean:
 	@bazel clean
 
 .PHONY: test
-test: build
-	@bazel test //...
+test:
+	@bazel test ${BUILD_FLAGS} //...
```
Comment (kyessenov) [inline Makefile]: I wonder: does this work fine on native mac OS? Should we restrict k8 to e2e test only?

File: README.md [hunk 2]
```diff
@@ -4,19 +4,64 @@
...
 
-If you are interested in contributing to the project, please take a look at the [build instructions](doc/build.md) and the [testing infrastructure](doc/testing.md).
+
+To learn how you can contribute to Istio Pilot, please
+see the Istio
+[contribution guidelines](https://github.com/istio/istio/blob/master/CONTRIBUTING.md).
+
+# Quick start
+
+1. *Install Bazel:* [Bazel 0.6.0](https://github.com/bazelbuild/bazel/releases/tag/0.6.0) or
```
Comment (kyessenov) [inline README.md]: 0.6.1 is the latest (some bug fixes). You can leave 0.6.0 here, I'm hoping 0.6.x series is somewhat backwards compatible.

File: bin/init.sh [hunk 3]
```diff
@@ -5,6 +5,14 @@ set -o nounset
 set -o pipefail
 set -x
 
+detected_OS=`uname -s 2>/dev/null || echo not`
```
Comment (kyessenov) [inline bin/init.sh]: I think we don't really need the entire build here. `fetch` is sufficient. I can try to see whether that works later.

File: doc/testing.md [hunk 4]
```diff
@@ -69,16 +69,7 @@ To run the complete set of tests, use the following commands:
...
-the kube config file on the host to platform/kube instead of symlinking it,
-and change the paths to minikube certs.
-
-    cp ~/.kube/config platform/kube/
-    sed -i 's!/Users/<username>!/home/ubuntu!' platform/kube/config
-
-Also, copy the same file to `/home/ubuntu/.kube/config` in the VM, and make
-sure that the file is readable to user `ubuntu`.
-
+_Note1_: If you are using OS X, append `--cpu=k8` to the bazel test command.
```
Comment (kyessenov) [inline doc/testing.md]: _ to build linux binaries _

File: Makefile (hunk 1 above)
Comment (rshriram) [reply Makefile]: Yeah, its a bit of pain when this test starts e2e as well. if we restrict k8s to e2e alone, then it makes instructions and stuff easier. I could probably even code up the refresh goldens here.

File: bin/init.sh (hunk 3 above)
Comment (rshriram) [reply bin/init.sh]: comment on the wrong file? :)..

File: Makefile (hunk 1 above)
Comment (rshriram) [reply Makefile]: Actually, it works perfectly on Mac. Except the proxy/envoy test. And you don't even need the build flags. Updated PR

File: doc/testing.md (hunk 4 above)
Comment (rshriram) [reply doc/testing.md]: No longer needed. bazel test //... will run unit tests on Mac. bin/e2e.sh will automatically compile for linux and push to docker hub.

Comment (rshriram) [reply doc/testing.md]: Don't you love shell scripts? ;)

**Pull request merged**